### ISSUE TITLE
Unable to connect to Blob storage from AKS

### ISSUE THREAD

**Issue by ifekxp** [Azure/azure-sdk-for-python#27953]
- **Package Name**: azure-storage-blob
- **Package Version**: 12.13.1
- **Operating System**: Debian 11 (bullseye), AKS Kubernetes 1.24.6
- **Python Version**: 3.10.5

**Describe the bug**
I am working on migrating our python app that is running on our Azure VM to AKS. The python app uploads files from VM to Azure Blob Storage. When I try to run the app from AKS after dockerizing it, it is unable to upload file by throwing following error:

 ```
2022-12-14 19:15:29 WARNING Unable to upload: Retry: 0 FileName: abc.zip Exception: 
<urllib3.connection.HTTPSConnection object at 0x7fde819cec50>: 
Failed to establish a new connection: [Errno -2] Name or service not known
```
Initially I was getting same error from my laptop, please refer to the old reported issue where I added my comment to the end. 

But after adding the correct certificate, I was able make it working from minikube.

However, When I push the same image to ACR and run it from AKS, I still get above error.

The code that is working from VM (without last cert verification parameter) and Minikube (with last parameter) is like this:

```
 file_size = os.path.getsize(file_name)
 content_settings=ContentSettings(content_type='application/zip')
 blob_client.upload_blob(data, blob_type="BlockBlob",length=file_size, 
      content_settings = content_settings, timeout=600, connection_timeout=60, 
      connection_verify=False)
```
  
I tried adding "connection_verify=certifi.where()" or "connection_verify='/etc/ssl/certs/ca-certificates.crt'" to the end but it has same result. These value also throws error inside Minikube.

**To Reproduce**
1. Prepare a python code to upload a file and dockerize it
2. Run it on AKS
3. It throws "Failed to establish a new connection: [Errno -2] Name or service not known" error after some time

**Expected behavior**
The app should be able to upload file to blob storage. 
If there is error, it should clearly show what is the issue, for example, whether it's a DNS or certificate issue

**Screenshots**

**Packages**

_Dockerfile:_
```
FROM python:3.10.5
RUN apt-get update && apt-get -y install ca-certificates

# certifcate
ADD xyz.cer /usr/local/share/ca-certificates/xyz.crt
RUN chmod 644 /usr/local/share/ca-certificates/xyz.crt && update-ca-certificates

# copy file
WORKDIR /usr/src/app

COPY requirements.txt ./
COPY app.py ./
RUN chmod 755 ./app.py

RUN pip3 install --no-cache-dir --trusted-host pypi.org --trusted-host files.pythonhosted.org pip --upgrade 
RUN pip3 install --no-cache-dir --trusted-host pypi.org --trusted-host files.pythonhosted.org -r requirements.txt
...
```

_requirements.txt:_
```
azure-core==1.25.0
azure-storage-blob==12.13.1
certifi==2022.6.15
requests==2.28.1
requests-oauthlib==1.3.1
urllib3==1.26.11
...
```

**DNS and connectivity  from container:**
```
pod$ # host <XYZ>.blob.core.windows.net
<XYZ>.blob.core.windows.net is an alias for <XYZ>.store.core.windows.net.
<XYZ>.store.core.windows.net has address 20....

pod$ curl -v https://<XYZ>.blob.core.windows.net/
*   Trying 20....
* Connected to <XYZ>.blob.core.windows.net (20....) port 443 (#0)
...
* SSL connection using TLSv1.2 / ...
* ALPN, server did not agree to a protocol
* Server certificate:
*  subject: CN=*.blob.core.windows.net
*  start date: Sep 22 08:13:11 2022 GMT
*  expire date: Sep 22 08:13:11 2023 GMT
*  subjectAltName: host "<XYZ>.blob.core.windows.net" matched cert's "*.blob.core.windows.net"
*  issuer: C=US; O=Microsoft Corporation; CN=Microsoft RSA TLS CA 01
*  SSL certificate verify ok.
> GET / HTTP/1.1
> Host: <XYZ>..blob.core.windows.net
> User-Agent: curl/7.74.0
> Accept: */*
> 
...
< HTTP/1.1 400 Value for one of the query parameters specified in the request URI is invalid.
< Content-Length: 351
< Content-Type: application/xml
< Server: Microsoft-HTTPAPI/2.0
```

**Additional context**

Reference:

[upload_blob()](https://learn.microsoft.com/en-us/python/api/azure-storage-blob/azure.storage.blob.blobclient?view=azure-python#azure-storage-blob-blobclient-upload-blob)

[Reported docker issue# 13363](https://github.com/Azure/azure-sdk-for-python/issues/13363)

**Comment by kristapratico** [comment]
Thanks for the detailed issue, we'll take a look as soon as possible.

**Comment by ifekxp** [comment]
I wrote a test app and was able to make it working. I will go back to my original app code and find out what caused it throwing that error. I will post if I find something related to sdk.

Test app does need “connection_verify=False” parameter. It will be interesting to know how I can pass certificates to upload method.

**Comment by jalauzon-msft** [comment]
Hi @ifekxp, thanks for checking back about your test app. Something that may help to specify your own cert is the additional `connection_cert` keyword. I believe this allows you to specify your own client-side cert. This is not well documented from the Storage SDK itself, but here is the documentation on it from `azure-core`. This gets passed in the same fasion as `connection_verify`. Note, we should be using the `certifi` cert by default.

https://github.com/Azure/azure-sdk-for-python/blob/04e4891c70a4297c015abb354696105632c08ff7/sdk/core/azure-core/azure/core/configuration.py#L99-L102

Hopefully this helps, thanks.

**Comment by evheniyt** [comment]
/unresolve

I have the same issue running `azure-storage-blob` on AKS. The app works fine locally with the same configuration, but when I'm trying to run it on AKS I'm getting:
```
ServiceRequestError('<urllib3.connection.HTTPSConnection object at 0x7f192e0823b0>: Failed to establish a new connection: [Errno -2] Name or service not known
```
I tried using different authentication methods (Access keys, SAS URL) but had the same error.
My connection config doesn't use any certificates and looks like this (for `Access keys` method)
```
blob_client = BlobServiceClient.from_connection_string(conn_str=connection_string)
```
I have also tried to add `connection_verify=False` but again, it works locally but doesn't work on AKS, with the same error.
From the pod, that is running the app I could resolve and access my storage account by using curl
